tests/x509/test_x509.py

```diff
@@ -3752,6 +3752,18 @@ def test_unsupported_subject_public_key_info(self, backend):
         with pytest.raises(ValueError):
             cert.public_key()
 
+    def test_bad_time_in_validity(self, backend):
+        cert = _load_cert(
+            os.path.join(
+                "x509", "badasn1time.pem"
+            ),
+            x509.load_pem_x509_certificate,
+            backend,
+        )
+
+        with pytest.raises(ValueError):
```

Comment:
Please add an assert that the message on the exception contains the expected value

Reply:
This can be done with `pytest.raises(ValueError, match='string')` btw

Reply:
I guess that value and contains are slightly ambiguous here.  To clarify, would you prefer something like this:

```
assert exc.value.message == "Couldn't parse ASN.1 time as generalizedtime '19020701025736Z'"
```

Or something like this:

```
assert exc.value.message.find('19020701025736Z') > -1
```

Reply:
The former please. Yuo can do `with pytest.raises(ValueError, match='Couldn't parse ASN.1 time as generalizedtime'):` and it will do a match on that.

Reply:
Oh, that's very nice.  I will push one more commit.